server/storage/mvcc/watchable_store.go

```diff
@@ -199,6 +202,7 @@ func (s *watchableStore) cancelWatcher(wa *watcher) {
 		time.Sleep(time.Millisecond)
 	}
 
+	wa.span.End()
```

Comment:
We should be careful to make sure that we don't leak any spans. Are you sure all watchers opened will be closed through `cancelWatcher`?

Reply:
Yes, `Watch` always calls `sws.close()` before returning which will go through all the `cancel` functions and invoke them: https://github.com/etcd-io/etcd/blob/v3.6.4/server/etcdserver/api/v3rpc/watch.go#L226

Alternatively, we can store `span` reference alongside `cancels` for the same effect (and maybe avoid modifying `watch`)